### PR TITLE
Add abuse reporting endpoint for extensions aka add-ons.

### PR DESCRIPTION
**Pull request by eviljeff**
Bonus: remove high abuse reporting trigger code that doesn't work.

### CODE REVIEW

File: mkt/abuse/models.py [hunk 1]
```diff
@@ -18,13 +19,15 @@ class AbuseReport(ModelBase):
     reporter = models.ForeignKey(UserProfile, null=True,
                                  blank=True, related_name='abuse_reported')
     ip_address = models.CharField(max_length=255, default='0.0.0.0')
-    # An abuse report can be for an addon, a user, or a website. Only one of
-    # these should be set.
+    # An abuse report can be for an app, a user, a website, or an extension.
+    # Only one of these should be set.
     addon = models.ForeignKey(Webapp, null=True, related_name='abuse_reports')
```
Comment (diox) [inline mkt/abuse/models.py]: Ah, this is not going to be confusing at all :-)

File: mkt/abuse/serializers.py [hunk 2]
```diff
@@ -52,3 +54,13 @@ class WebsiteAbuseSerializer(BaseAbuseSerializer):
 
     class Meta(BaseAbuseSerializer.Meta):
         fields = BaseAbuseSerializer.Meta.fields + ('website',)
+
+
+class ExtensionAbuseSerializer(BaseAbuseSerializer):
+    extension = SplitField(
+        SlugOrPrimaryKeyRelatedField(source='extension', slug_field='slug',
+                                     queryset=Extension.objects.all()),
```
Comment (diox) [inline mkt/abuse/serializers.py]: You probably want `Extension.objects.without_deleted().public()` as the queryset instead. I'm surprised this isn't done like this for apps/websites too.

Comment (eviljeff) [inline mkt/abuse/serializers.py]: Can you log an abuse report for a deleted <thing> ?  Also, no slugs for deleted, is that a problem?  I had serializers.PrimaryKeyRelatedField at first btw - I think Fireplace just uses the pk.

File: mkt/abuse/models.py (hunk 1 above)
Comment (eviljeff) [inline mkt/abuse/models.py]: I thought about changing it but there would be a likely cascade of addon->app changes needed...

Comment (diox) [inline mkt/abuse/models.py]: Yeah, definitively not the right time for this

File: mkt/abuse/serializers.py (hunk 2 above)
Comment (diox) [inline mkt/abuse/serializers.py]: My point was that you shouldn't be able to report non-public websites, hence my queryset suggestion. So I don't think using the slug is going to be an issue.

Comment (eviljeff) [inline mkt/abuse/serializers.py]: oh, I forgot Extensions was the other way around - deleted included in all()

Comment (eviljeff) [inline mkt/abuse/serializers.py]: changed

File: mkt/abuse/tests/test_views.py [hunk 3]
```diff
@@ -170,3 +176,28 @@ def test_invalid_website(self):
...
+            'sprout': 'potato',
+            'extension': self.extension.pk
+        }
+
+    def test_invalid_app(self):
+        res, data = self._call(data={'extension': -1})
+        eq_(400, res.status_code)
+        assert 'does not exist' in data['extension'][0]
+
+    def test_slug_app(self):
```
Comment (diox) [inline mkt/abuse/tests/test_views.py]: method name contains "app"

File: mkt/abuse/tests/test_views.py [hunk 4]
```diff
@@ -170,3 +176,28 @@ def test_invalid_website(self):
...
+        self.extension = Extension.objects.create(
+            name=u'Test Êxtension')
+        self.extension.update(status=STATUS_PUBLIC)
+        self.default_data = {
+            'text': 'Lies! This extension is an add-on!',
+            'sprout': 'potato',
+            'extension': self.extension.pk
+        }
+
+    def test_invalid_app(self):
```
Comment (diox) [inline mkt/abuse/tests/test_views.py]: method name contains "app"

Comment (diox) [inline mkt/abuse/tests/test_views.py]: can you also add a test with a deleted extension ?

Comment (eviljeff) [inline mkt/abuse/tests/test_views.py]: done

**Comment by diox** [comment]
r+

**Pull request merged**